src/pages/VaccineAdminPage.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { ToggleBar, DataTablePageView, SearchBar, PageButton } from '../widgets';
+import { DataTable, DataTableRow, DataTableHeaderRow } from '../widgets/DataTable';
+
+import { recordKeyExtractor, getItemLayout, getPageDispatchers } from './dataTableUtilities';
+
+import { ROUTES } from '../navigation';
+import { generalStrings, vaccineStrings } from '../localization';
+import globalStyles from '../globalStyles';
+
+export const VaccineAdminPageComponent = ({
+  data,
+  sortKey,
+  isAscending,
+  columns,
+  dataSet,
+  modalKey,
+  dataState,
+  searchTerm,
+  onFilterData,
+  onEditLocationCode,
+  onEditLocationDescription,
+  onSortColumn,
+  onToggleFridges,
+  onToggleSensors,
+}) => {
+  const getCallback = colKey => {
+    switch (colKey) {
+      case 'description':
+        return onEditLocationDescription;
+      case 'code':
+        return onEditLocationCode;
+      default:
+        return null;
+    }
+  };
+
+  const renderHeader = React.useCallback(
+    () => (
+      <DataTableHeaderRow
+        columns={columns}
+        onPress={onSortColumn}
+        isAscending={isAscending}
+        sortKey={sortKey}
+      />
+    ),
+    [sortKey, isAscending, columns]
+  );
+
+  const renderRow = React.useCallback(
+    listItem => {
+      const { item, index } = listItem;
+
+      const rowKey = recordKeyExtractor(item);
+      return (
+        <DataTableRow
+          rowData={data[index]}
+          rowState={dataState.get(rowKey)}
+          rowKey={rowKey}
+          columns={columns}
+          getCallback={getCallback}
+          rowIndex={index}
+        />
+      );
+    },
+    [data, dataState]
+  );
+
+  const toggles = React.useMemo(
+    () => [
+      { text: vaccineStrings.fridges, onPress: onToggleFridges, isOn: dataSet === 'fridges' },
+      { text: vaccineStrings.sensors, onPress: onToggleSensors, isOn: dataSet === 'sensors' },
```

Comment:
Would like these to be constants, but a little bit .. whatever.. was considering making a `DataSet` or something reducer to manage this but was a bit lazy